src/tests/neighborhood_api.test.ts

```diff
@@ -26,6 +31,44 @@ describe('When neighborhoods already exist in the db', () => {
     // expect(response.body.length).toEqual(3);  // 3 was a magic number.
     expect(response.body.length).toEqual(numberOfNeighborhoods);
   });
+
+  test('DELETE /neighborhoods/:id removes a neighborhood', async () => {
+    const initialNeighborhoods = await api.get('/api/neighborhoods');
+    const loginResponse = await api
+      .post('/api/login')
+      .send(loginData);
+    const { token } = loginResponse.body;
+
+    const response = await api.delete('/api/neighborhoods/1').set('Authorization', `Bearer ${token}`);
+    const currentNeighborhoods = await api.get('/api/neighborhoods');
+
+    expect(response.status).toEqual(200);
+    expect(currentNeighborhoods.body.length).toEqual(initialNeighborhoods.body.length - 1);
+  });
+
+  test('User cannot delete neighborhood if he/she is not admin', async () => {
+    const initialNeighborhoods = await api.get('/api/neighborhoods');
+    const loginResponse = await api
+      .post('/api/login')
+      .send(loginData);
+    const { token } = loginResponse.body;
+
+    const response = await api.delete('/api/neighborhoods/2').set('Authorization', `Bearer ${token}`);
+    const currentNeighborhoods = await api.get('/api/neighborhoods');
+
+    expect(response.status).toEqual(403);
+    expect(currentNeighborhoods.body.length).toEqual(initialNeighborhoods.body.length);
+  });
+
+  test('User cannot delete neighborhood if he/she is not logged in', async () => {
+    const initialNeighborhoods = await api.get('/api/neighborhoods');
+
+    const response = await api.delete('/api/neighborhoods/2');
+    const currentNeighborhoods = await api.get('/api/neighborhoods');
+
+    expect(response.status).toEqual(401);
+    expect(currentNeighborhoods.body.length).toEqual(initialNeighborhoods.body.length);
+  });
```

Comment:
I am not sure whether this way of writing test confirms to best practices. We are making the test of `DELETE /neighborhoods/:id` dependent on the working of `GET /api/neighborhood` and other routes. If this test fails, then we are not sure the problem is due to which route.

I think, within tests, we should try to use prisma queries as much as possible and not make our tests for routes dependent on other routes.

Now this will get complicated as the data returned by routes becomes more complicated due to JOIN queries. In that case I dont know what the best practices are.

If you want I can try rewriting the tests for these routes.

Reply:
I understand your point and I agree. If we can get by with using Prisma queries directly, we should do that.
Yeah, I'd love to get some help with this. Just push the changes directly to this branch.